frontend/public/components/storage-class-form.tsx

```diff
@@ -0,0 +1,927 @@
+    const nameUpdated = updatedName !== this.previousName;
+    const returnVal = {
+      error: null,
+      nameIsValid: true
+    };
+
+    if (nameUpdated) {
+      if (updatedName.trim().length === 0) {
+        returnVal.error = 'Storage name is required';
+        returnVal.nameIsValid = false;
+      } else if (this.resources) {
+        for (let i = 0; i < this.resources.data.length; i++) {
+          if (this.resources.data[i].metadata.name === updatedName.toLowerCase()) {
+            returnVal.error = 'Storage name must be unique';
+            returnVal.nameIsValid = false;
+          }
+        }
+      }
+
+      this.previousName = updatedName;
+    }
+
+    return returnVal;
+  };
+
+  validateParameters = () => {
+    let validationSuccessful = true;
+    const params = this.state.newStorageClass.parameters;
+    for (let key in params) {
+      if (params.hasOwnProperty(key) && params[key].validationMsg !== null) {
+        validationSuccessful = false;
+        break;
+      }
+    }
+
+    return validationSuccessful;
+  };
+
+  allRequiredFieldsFilled = () => {
+    if (this.state.newStorageClass.name.trim().length === 0) {
+      return false;
+    }
+
+    const storageType = this.state.newStorageClass.type;
+    if (storageType === null) {
+      return false;
+    }
+
+    const allParamsForType = this.storageTypes[storageType].parameters;
+    const userEnteredParams = this.state.newStorageClass.parameters;
+    let allReqdFieldsEntered = true;
+
+    for (let key in allParamsForType) {
+      if (allParamsForType.hasOwnProperty(key)) {
+        const value = _.get(userEnteredParams, `${key}.value`, null);
+        const valueExists = value !== null && value !== '' && !_.isEmpty(value);
+
+        if (this.paramIsRequired(key) && !valueExists) {
+          allReqdFieldsEntered = false;
+          break;
+        }
+      }
+    }
+
+    return allReqdFieldsEntered;
+  };
+
+  paramIsRequired = (paramKey, params = this.state.newStorageClass.parameters) => {
+    const requiredParam = _.get(this.storageTypes[this.state.newStorageClass.type], ['parameters', paramKey, 'required'], null);
+    let isRequired = false;
+    if (requiredParam) {
+      isRequired = typeof requiredParam === 'function' ? requiredParam(params) : requiredParam;
+    }
+
+    return isRequired;
+  };
+
+  getProvisionerElements = () => {
+    const dynamicContent = Object.keys(this.storageTypes[this.state.newStorageClass.type].parameters).map(key => {
+      const parameter = this.storageTypes[this.state.newStorageClass.type].parameters[key];
+      const validationMsg = _.get(this.state.newStorageClass.parameters[key], 'validationMsg', null);
+      const isCheckbox = parameter.type === 'checkbox';
+
+      if (parameter.visible && !parameter.visible(this.state.newStorageClass.parameters)) {
+        return null;
+      }
+
+      let children =
+        <React.Fragment>
+          <FormControl
+            type={isCheckbox ? 'checkbox': 'text'}
```

Comment:
The checkboxes didn't seem to render correctly for me when I tried them.

<img width="359" alt="screen shot 2018-10-10 at 8 39 08 am" src="https://user-images.githubusercontent.com/1167259/46736883-1ec9b200-cc68-11e8-8d79-6bf55386f873.png">

Reply:
Fixed